client/internal/clientcommon.go

```diff
@@ -191,12 +192,28 @@ func (c *ClientCommon) SetAgentDescription(descr *protobufs.AgentDescription) er
 // calcHashEffectiveConfig calculates and sets the Hash field from the rest of the
 // fields in the message.
 func calcHashEffectiveConfig(msg *protobufs.EffectiveConfig) {
+	cfgMap := msg.GetConfigMap().GetConfigMap()
+
+	// Construct hash
 	h := sha256.New()
-	if msg.ConfigMap != nil {
-		for k, v := range msg.ConfigMap.ConfigMap {
-			h.Write([]byte(k))
-			h.Write(v.Body)
-			h.Write([]byte(v.ContentType))
+
+	// If the config is empty don't attemp to add more to the hash
+	if cfgMap != nil || len(cfgMap) > 0 {
```

Comment:
So if I see correctly, `cfgMap` is `map[string]*protobufs.AgentConfigFile{}`, which means you don't need to check for nil, i.e.

```suggestion
	if len(cfgMap) > 0 {
```

Reply:
Yeah you're right. I forgot the `len` returns 0 for nil or empty.